test/resources/qemu-guest-agent.py

```diff
@@ -0,0 +1,302 @@
+    if not vm_name:
+        raise RuntimeError('vm name is not specified')
+    if not pid:
+        raise RuntimeError('pid is not specified')
+
+    agent_result = get_process_result(vm_name, pid)
+    while not agent_result['exited']:
+        if agent_result['exitcode'] is not None:
+            break
+        if timeout is not None:
+            timeout -= 1
+            if timeout <= 0 and not agent_result['exited']:
+                if on_timeout == 'terminate':
+                    terminate_process(vm_name, pid)
+                    BuiltIn().log(f'process {pid} on {vm_name} timed out, terminating', 'WARN')
+                elif on_timeout == 'kill':
+                    terminate_process(vm_name, pid, kill=True)
+                    BuiltIn().log(f'process {pid} on {vm_name} timed out, killing', 'WARN')
+                elif on_timeout == 'continue':
+                    BuiltIn().log(f'process {pid} on {vm_name} timed out, continuing', 'WARN')
+                    break
+                else:
+                    raise RuntimeError(f'invalid on_timeout value: {on_timeout}')
+                break
+            BuiltIn().sleep(1)
+        agent_result = get_process_result(vm_name, pid)
+
+    return agent_result
+
+
+def run_process(vm_name: str, cmd: str, *args, timeout: int = 0, on_timeout: str = None) -> (dict, bool):
+    """
+    :param vm_name:     The name of the VM to execute the command on
+    :type vm_name:      str
+    :param cmd:         The absolute path to a command on the guest, e.g. "/bin/ls"
+    :type cmd:          str
+    :param args:        The arguments to pass to the command, separated into a list of strings, e.g. ["-l", "/tmp"]
+    :type args:         list
+    :param timeout:     The maximum amount of time to wait for the process to complete, in seconds
+    :type timeout:      int
+    :param on_timeout:  The action to take if the timeout is reached. See the RF Process.wait_for_process
+                        documentation for more information.
+    :type on_timeout:   str
+    :return:            The stdout, stderr, and exit code of the guest process for the given PID
+    :rtype:             (dict, bool)
+    :raises             RuntimeError: If the VM name or command path is not specified
+
+    ``run_process`` excepts a given command and optional arg list to execute on a given VM using the
+    qemu-agent-command. The command is executed by using virsh qemu-guest-command. This is a blocking call unless
+    qemu-agent-timeout is set to anything but the default.  See the guest-agent-timeout section of man virsh for info
+    on timeout configuration.
+
+    IMPORTANT: Using `/bin/bash` as the cmd is highly discouraged. This is because the qemu-agent-command
+    will execute the command in a subshell on the guest, and return immediately. The PID returned by qemu-agent-command
+    is the PID of the subshell, not the command executed within the subshell. This is a known limitation of the
+    qemu-agent.
```

Comment:
Am I right that means don't run commands like `/bin/bash -c do-something`, but it is OK for the command to be a shell script?

Reply:
I swear I read this in documentation, but ofc cannot find it.  Some ad hoc testing with waits, forced exits, and complex commands seem pretty safe.  I'll drop this comment.